contracts/protocol/pool/PoolCore.sol

```diff
@@ -528,6 +529,61 @@ contract PoolCore is
         );
     }
 
+    /// @inheritdoc IPoolCore
+    function swapPToken(
+        address srcAsset,
+        uint256 srcAmount,
+        address dstAsset,
+        address to,
+        bytes32 swapAdapterId,
+        bytes calldata swapPayload
+    ) external nonReentrant {
+        DataTypes.PoolStorage storage ps = poolStorage();
+
+        SwapLogic.executeSwapPToken(
+            ps,
+            ps._usersConfig[msg.sender],
+            DataTypes.ExecuteSwapParams({
+                srcAsset: srcAsset,
+                dstAsset: dstAsset,
+                amount: srcAmount,
+                user: to,
+                reservesCount: ps._reservesCount,
+                oracle: ADDRESSES_PROVIDER.getPriceOracle(),
+                priceOracleSentinel: ADDRESSES_PROVIDER.getPriceOracleSentinel(),
+                swapAdapter: ps._swapAdapters[swapAdapterId],
+                swapPayload: swapPayload
+            })
+        );
+    }
+
+    /// @inheritdoc IPoolCore
+    function swapDebt(
+        address srcAsset,
+        uint256 srcAmount,
+        address dstAsset,
+        bytes32 swapAdapterId,
+        bytes calldata swapPayload
```

Comment:
is there any risk that's introduced by this swap payload? or can we consider it safe?